src/dlc-handlers/software-wallet-dlc-handler.ts

```diff
@@ -212,9 +224,45 @@ export class SoftwareWalletDLCHandler {
         vault.btcFeeRecipient,
         vault.btcRedeemFeeBasisPoints.toBigInt()
       );
-      return Transaction.fromPSBT(closingTransaction);
+      return withdrawalTransaction;
     } catch (error: any) {
-      throw new Error(`Error creating Closing PSBT: ${error}`);
+      throw new Error(`Error creating Withdrawal PSBT: ${error}`);
+    }
+  }
+
+  async createDepositPSBT(
+    depositAmount: bigint,
+    vault: RawVault,
+    attestorGroupPublicKey: string,
+    fundingTransactionID: string,
+    feeRateMultiplier?: number,
+    customFeeRate?: bigint
+  ) {
+    const { fundingPayment, multisigPayment } = await this.createPayments(
+      vault.uuid,
+      attestorGroupPublicKey
+    );
+
+    if (multisigPayment.address === undefined || fundingPayment.address === undefined) {
+      throw new Error('Payment Address is undefined');
     }
+
+    const feeRate =
+      customFeeRate ??
+      BigInt(await getFeeRate(this.bitcoinBlockchainFeeRecommendationAPI, feeRateMultiplier));
```

Comment:
I wonder if you can DRY up these lines from the 3 kinds of createDepositPSBT functions. They seem almost 100% similar. Perhaps a `prepareDepositPSBT` helper function somewhere?